controllers/vrg_volsync.go

```diff
@@ -425,3 +425,24 @@ func (v *VRGInstance) disownPVCs() error {
 
 	return nil
 }
+
+// cleanupResources this function deleted all PS, PD and VolumeSnapshots from its owner (VRG)
+func (v *VRGInstance) cleanupResources() error {
+	for idx := range v.volSyncPVCs {
```

Comment:
We iterate on the PVCs - but we delete other resources (replication source/destination, volume snapshots). Does volSyncPVCs change after we delete? If one delete call fails below, do we run again for all PVCs on the next try?

Reply:
The reason we are iterating on the PVCs is that we need the PVC name and namespace in order to locate and delete RD, RS and snapshots. It is done upon VRG deletion, so all the resources, owned by VRG must be deleted or disowned. If we fail to delete one of the resources, VRG can't be deleted (we don't want to have orphan resources). So, as I see it, nothing will happen to volSyncPVCs  until all owned resource will be eliminated, so next time we will run again on all PVCs